server_addon/max/client/ayon_max/api/lib.py

```diff
@@ -284,6 +285,39 @@ def reset_frame_range(fps: bool = True):
         frame_range["frameStartHandle"], frame_range["frameEndHandle"])
 
 
+def get_fps_for_current_context():
+    """Get fps that should be set for current context.
+
+    Todos:
+        - Skip project value.
+        - Merge logic with 'get_frame_range' and 'reset_scene_resolution' ->
+            all the values in the functions can be collected at one place as
+            they have same requirements.
+
+    Returns:
+        Union[int, float]: FPS value.
+    """
+    task_entity = get_current_task_entity(fields={"attrib"})
+    fps = task_entity.get("attrib", {}).get("fps")
+    if not fps:
+        project_name = get_current_project_name()
+        folder_path = get_current_folder_path()
+        folder_entity = ayon_api.get_folder_by_path(
+            project_name, folder_path, fields={"attrib.fps"}
+        ) or {}
+
+        fps = folder_entity.get("attrib", {}).get("fps")
+        if not fps:
+            project_entity = ayon_api.get_project(
+                project_name, fields=["attrib.fps"]
+            ) or {}
+            fps = project_entity.get("attrib", {}).get("fps")
+
+            if not fps:
+                fps = 25
+    return fps
```

Comment:
As far as I know in AYON task entities should **always** have the FPS attrib so there should never be a need for a fallback? I've asked this a few times and I believe @iLLiCiTiT confirmed that before.

But if that's the case - this would just be:
```python
task_entity = get_current_task_entity(fields={"attrib"})
return task_entity["attrib"]["fps"]
```

Reply:
Task will always have `"fps"`.